src/vss_tools/vspec/vssexporters/vss2apigear.py

```diff
@@ -0,0 +1,522 @@
+
+            for element in node.allowed:
+                element_str = str(element)
+                element_dec = hex_to_dec(element_str)
+                variant = get_allowed_enum_literal(element_str)
+                log.debug(f"  New variant: {variant}")
+                if VSSDataType.is_numeric(node.datatype) and is_floating_point(node.datatype):
+                    enum.variants.append((variant, None))
+                else:
+                    enum.variants.append((variant, element_dec))
+
+            property = ApiGearProperty(ApiGearType(enum_name))
+            if node.description != "":
+                property.description = node.description
+            if interface is not None:
+                interface.properties[node.name] = property
+                module.enumerations.append(enum)
+        else:
+            if node.has_datatype() and node.datatype is not None:
+                datatype = node.datatype.value
+            elif hasattr(node, "data_type_str"):
+                datatype = node.data_type_str
+            else:
+                log.warning(f"Node {node.qualified_name()} does not have a datatype")
+                return
+            prop = generate_property(node, datatype)
+            if prop is not None and interface is not None:
+                interface.properties[node.name] = prop
+
+
+def export_data_type_node(node: VSSNode | None, module: ApiGearModule, structure: ApiGearStructure | None = None):
+    """This method is used to traverse through the root VSS Data Types node to build
+       -> ApiGear equivalent string buffer and to serialize it accordingly into a module
+    """
+    if node is None:
+        return
+
+    log.debug(f"Node [{node.type}] {node.qualified_name()}")
+
+    if node.type == VSSType.BRANCH:
+        for child in node.children:
+            export_data_type_node(child, module)
+    elif node.type == VSSType.STRUCT:
+        struct_name = node_name(node)
+        struct = ApiGearStructure(struct_name)
+        log.debug(f"Creating a new structure {struct_name}")
+
+        for child in node.children:
+            export_data_type_node(child, module, struct)
+
+        module.structures.append(struct)
+    else:
+        if node.has_datatype() and node.datatype is not None:
+            datatype = node.datatype.value
+        elif hasattr(node, "data_type_str"):
+            datatype = node.data_type_str
+        else:
+            log.warning(f"Node {node.qualified_name()} does not have a datatype")
+            return
+        prop = generate_property(node, datatype)
+        if prop is not None and structure is not None:
+            structure.properties[node.name] = prop
+
+
+def generate_solution(directory: Path, module_filename: str, module_name: str,
+                      layers: typing.Dict[SolutionLayers, Path]):
+    log.info("Generating solution file")
+
+    path = os.path.join(directory, SOLUTION_FILENAME.format(module_name))
+    file = open(path, "w")
```

Comment:
file descriptor is never closed.
Use a context manager:
```python
with open (path, "w") as f:
    f.write()
```

Reply:
Reworked this function completely so it no longer has a never closed file descriptor.